### DISCUSSION TITLE
Help with installation on Ubuntu 22.04 server (local installation)

### DISCUSSION THREAD

**Discussion by mike-ua** [getredash/redash#6365] [Q&A]
Hi,
I could really use some help installing Redash - I've already make 3-4 attempts (at 3- 4 hours each) and no luck.

Local installation, personal use on Ubuntu 22.04 (updated before installation).

The last I tried was downloading the script from Github ($wget https://github.com/getredash/setup.git), and that did some things which ended in 'Saving to: setup.git' plus a confirmation it was saved.

But there are no further instructions anywhere, so I tried:
just entering 'setup.git' > command not found
cd'ing to 'setup.git' > Not a directory

What are the next steps?  Maybe this can be added to the 'Readme.md'?

Thanks.
Mike

**Comment by justinclift** [comment]
Ahhh.  Try downloading this file: https://github.com/getredash/setup/raw/master/setup.sh

That should give you a shell script (text file), that's just a sequence of commands to be run in order to setup Redash.  You should be able to just double click on the downloaded script (or run it from the command line), and it'll go ahead and install Redash.

That being said, I'm not personally sure if it runs ok on Ubuntu 22.04 LTS, as haven't tried the script on that yet.  The script at least *used* to work fine on Ubuntu 20.04 LTS though.

We'll likely go through that "setup" GitHub repository in a week or so, in order to get it updated / definitely working with newer Ubuntu versions.  And maybe newer versions of other Linux distributions too.

But, that task is behind some other ones presently being worked on in the main "redash" repo.

**Reply by mike-ua** [reply]
Hi Justin,
thanks for your reply!
Ok, I downloaded the file you listed, same thing, so then I understood I needed to learn how to run a script.  So Googl'ed and came up with this:
Step 1: 
chmod +x setup.sh
Step 2:
./setup.sh
Now I could clearly see the server connecting, downloading and unpacking - partial success!!

Next, per the instructions under 'Setup' on this page https://redash.io/help/open-source/setup#docker
I tried to: "Redash should be available using the server IP or DNS name you assigned to it. You can point your browser to this address."

I entered the server IP address (found with '$ ip addr show') in the web browser of the laptop I'm using to connect to the server but it did not find anything.

My guess is that I need to set up a user (for my laptop) on Ubuntu Server?  Or some permissions are getting in the way?  Or do I need to assign a 'DNS name' to Redash? (I have my laptop connected directly to the Server computer by LAN cable, no router).

Thanks,
Mike

**Reply by justinclift** [reply]
> I entered the server IP address (found with '$ ip addr show') in the web browser of the laptop I'm using to connect to the server but it did not find anything.

Hmmm, it sounds like the setup *probably* worked ok, but there might be a routing issue.

First things first, let's run the "docker-compose" command which checks if all the necessary pieces are up and running.  It needs to be run in the `/opt/redash` directory to work properly.  So:

```
cd /opt/redash
docker-compose ps
```

Here's the output on one of my systems just now, to show you what a working system looks like:

```
# docker-compose ps
          Name                         Command                  State                Ports          
----------------------------------------------------------------------------------------------------
redash_adhoc_worker_1       /app/bin/docker-entrypoint ...   Up             5000/tcp                
redash_postgres_1           docker-entrypoint.sh postgres    Up (healthy)   5432/tcp                
redash_redis_1              docker-entrypoint.sh redis ...   Up (healthy)   6379/tcp                
redash_scheduled_worker_1   /app/bin/docker-entrypoint ...   Up             5000/tcp                
redash_scheduler_1          /app/bin/docker-entrypoint ...   Up             5000/tcp                
redash_server_1             /app/bin/docker-entrypoint ...   Up             127.0.0.1:5000->5000/tcp
redash_worker_1             /app/bin/docker-entrypoint ...   Up             5000/tcp
```

The `State` column needs to say `Up` for all of them, and for the ones that have extra detail (eg "healthy") it needs to say "healthy" and not anything else.  Especially not "restarting", which would mean there's a problem. :smile:

Oh, you might have an `redash_nginx_1` entry as well (and maybe something "email" related too).  Mine won't have that listed, as the production setup we run is a bit customised.

---

If your setup is showing everything is working ok, then my suspicion is you're having a networking / routing problem.  This is actually sounding super familiar, but I'm just not recalling the specifics of the problem.

Hmmm, maybe something like the server only listening on it's localhost interface? (from super rough memory)

**Reply by mike-ua** [reply]
Ok I ran 
docker-compose ps
all is good (all say 'up', no 'extra' information')

could it be the firewall (ufw) on the server blocking my laptop? Maybe permissions need to be given?

Just fyi, I have a clean install of Ubuntu server, nothing else on it except Redash, and this is the first time I have ever tried to connect it with my laptop (via direct LAN cable), so I have not set up a local network yet.  The laptop is running Solus OS (/Budgie), latest updates.  I tried connecting to the server from the laptop via the file manager, I tried all prefix options (ftp://(server ip), ftps://(server ip), nfs://(server ip), etc. none worked.

Also tried adding the laptop ip to the server via
sudo ip addr add ...
but it could not find device " " (tried many options for the 'dev' argument, none worked, most likely I did not add a correct one).
Thanks.

**Reply by justinclift** [reply]
> all is good (all say 'up', no 'extra' information')

Cool.  Yeah, that means it installed Redash fine and is waiting to be configured.  That'll be listening on the http:// address on the server.

> could it be the firewall (ufw) on the server blocking my laptop?

Yeah, that *might* be the problem.  I don't personally use ufw, as I use a different firewall product so that'll be something for you to look into.

My *suspicion* is that if you can run the web browser on your server directly, then you'd probably be able to connect to the Redash server using http://localhost/.  That's just a suspicion though.  I need to take a proper look at the whole "setup.sh" process in detail (likely next week sometime) to refresh my memory (etc) though. :frowning:

**Reply by mike-ua** [reply]
ufw comes bundled with Ubuntu server.  I'll look into the configurations.
Also, I'll check on the Solus side - I don't think it really would be that different (it is Linux, after all), but maybe there is something on the forums there related to networking.

ok, than I guess one last question for now - how would I "run the web browser on your server directly"?  I didn't think that was possible on a Server edition of Ubuntu (e.g. no gui).  Maybe as a snap it can do this today?  And should I attempt this?

**Reply by justinclift** [reply]
> how would I "run the web browser on your server directly"?

Weeeeeeeeeeell, I'm assuming that this is an experimental setup (for now) that you're trying things out on.

For an experimental setup, you can probably do the standard `sudo apt install firefox` (or equivalent).

Doing that will probably download and install a bazillion dependencies (like half the desktop packages, etc :grin:).  Not what you want for a production server, but for an internal-only thing you're mucking around with it'd be fine.

If you have X forwarding enabled in your remote server config, then you should be able to ssh into it and start up firefox (on the server)... which will then be displayed across the network onto your local computer (all going through the ssh connection).

Alternatively, because X forwarding can be a pita to set up... you could potentially run something like VNC (eg https://tigervnc.org) on the server and log into a desktop session using that.  Then launch Firefox on that desktop session).

*If* you go down any of those routes, then I'd recommend just doing this firefox/web browser stuff just to see if it works for letting you access the Redash web interface.  eg as a diagnostic step.

If this server is running on a publicly accessible internet address, then I'd very much recommend NOT taking this approach.  It's insecure as all hell.  *Definitely* do not put in real database access info (url, username, pw) that you've set up in this way.

**Reply by mike-ua** [reply]
ok, it sounds like I'm going down the 'wrong' path.

I just assumed Redash worked on all Ubuntu server editions, but I see it's only tested on 18.04 (which is EOL and you are now addressing).

First, I completely forgot about Webmin or Cockpit which I should definitely use as I am not a server admin nor is that my direction.
Next, I should first set up and make sure the network (really, it's just a server and one laptop) works correctly via these interfaces.
Then  re-test Redash based on Ubuntu 22.04 (even though it looks like you will only go to 20.04 from 18.04) and provide feedback.
If all the above fails, I will wipe the server, and re-do all these steps with Ubuntu 20.04.  18.04 is no long an option as it's a security risk.

Really, all I want to be is a Data Analyst  - but there are very few options for Linux users for Dashboarding tools (even paying for Tableau or MS Power BI is not an option).  My open source faith is being tested after leaving the MS world 15 years ago.

**Reply by justinclift** [reply]
Yeah.  The project is currently in a (temporary) not-great-state for deployment purposes, as after the Databricks acquihire of the original team things kind of fell in a heap (mostly).

Since switching over to being a Community Led project about um.... a month and a half ago (roughly), we've been madly merging PRs, updating Dependabot alerts, fixing various immediately pressing problems, etc.  This "update all the things before making a release" piece isn't yet finished, though we're coming to the likely tail end of it.

We'll be making an initial release some time in the next few weeks, after a period of testing, and likely some alpha/beta releases (tbd).  We clearly also need to address the whole "how to deploy this in production?" side of things about that time too.

You just happen to be coming to the project when these above bits aren't completed yet, and so everyone's been focused on the earlier-stage pieces. :grin:

Nothing wrong with that, but yeah... it's all a bit of a mess at the moment for people wanting to deploy it. :wink:

**Reply by mike-ua** [reply]
well, maybe I can help out - but not sure I would be too valuable yet, my Python is at a very low level.  I'm not sure I'm ready for bug fixes but who knows, maybe bugs that are super easy but time consuming for a more experienced dev?

But I guess the more immediate way of helping is just installing, testing and providing feedback.  Maybe it would be more useful if I wiped the server, re-installed with 20.04 and tested that?  Can both be installed on the same server for testing?

**Reply by justinclift** [reply]
> Maybe it would be more useful if I wiped the server, re-installed with 20.04 and tested that?

Yeah, that's the better approach.  20.04 is known to be good, so best to work with (for now).

Personally, I'd recommend you experiment with stuff in a local VM on your laptop/desktop, rather than a remote server (for now).  That'll let you try things out in an easier fashion, but it's up to you of course. :smile:

**Reply by mike-ua** [reply]
ok, some people at the Solus OS forums kindly reminded me that I need an Ethernet switch - I completely forgot our set-up of 3 computers and a server from many years ago.  So a hardware issue, not software most likely.  I'll get a switch (in between the server and laptop), correctly set up the server then test again.

**Reply by mike-ua** [reply]
I got an Ethernet switch, tested for 9 hours over two days, nothing worked - but the fact that the server was not seeing 'Cockpit' either is another indication the problem is with my server, not Redash.  Just like trying to find the problem with a failed operating system, it's quicker to just wipe and re-install, so I'll do the same with my server and start from zero.

**Reply by justinclift** [reply]
Sounds like a plan. :smile:
